#!/usr/bin/env python

PACKAGE = "cob_base_velocity_smoother"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

feedback_enum = gen.enum([gen.const("NONE",     int_t, 0, "No feedback"),
                          gen.const("ODOMETRY", int_t, 1, "Odometry feedback"),
                          gen.const("COMMANDS", int_t, 2, "Command feedback")],
                          "enum types robot velocity feedback")
gen.add("robot_feedback", int_t,  0, "What source to use as robot velocity feedback", 0, None, None, edit_method=feedback_enum)

gen.add("speed_lim_vx", double_t, 0, "Maximum linear velocity", 1.0, 0.0, 10.0)
gen.add("speed_lim_vy", double_t, 0, "Maximum linear velocity", 1.0, 0.0, 10.0)
gen.add("speed_lim_w",  double_t, 0, "Maximum angular velocity", 1.0, 0.0, 10.0)

gen.add("accel_lim_vx", double_t, 0, "Maximum linear acceleration", 0.7, 0.0, 10.0)
gen.add("accel_lim_vy", double_t, 0, "Maximum linear acceleration", 0.7, 0.0, 10.0)
gen.add("accel_lim_w",  double_t, 0, "Maximum angular acceleration", 0.7, 0.0, 10.0)

gen.add("decel_factor", double_t, 0, "Deceleration to acceleration ratio", 2.0, 0.0, 10.0)
gen.add("safe_factor",  double_t, 0, "Multiplier in case safety stop is required", 4.0, 0.0, 10.0)

# Second arg is node name it will run in (doc purposes only), third is generated filename prefix
exit(gen.generate(PACKAGE, "velocity_smoother_configure", "params"))
